training/src/anemoi/training/diagnostics/plots.py

```diff
@@ -229,15 +229,15 @@ def plot_power_spectrum(
     grid_pc_lon, grid_pc_lat = np.meshgrid(regular_pc_lon, regular_pc_lat)
 
     for plot_idx, (variable_idx, (variable_name, output_only)) in enumerate(parameters.items()):
-        yt = y_true[..., variable_idx].squeeze()
-        yp = y_pred[..., variable_idx].squeeze()
+        yt = (y_true if y_true.ndim == 1 else y_true[..., variable_idx]).reshape(-1)
```

Comment:
@lzampier - why do you need the if here - could just do y_true[..., variable_idx]).reshape(-1)? and same for x

Reply:
Probably yes. I was overcautious. Let me test later.